wsagent/che-core-api-project/src/test/java/org/eclipse/che/api/project/server/ProjectServiceTest.java

```diff
@@ -928,6 +939,50 @@ public void testDeleteProject() throws Exception {
     }
 
     @Test
+    public void testDeleteProjectsConcurrently() throws Exception {
+        int threadNumber = 100;
+        ExecutorService executor = Executors.newFixedThreadPool(threadNumber);
+        CountDownLatch countDownLatch = new CountDownLatch(threadNumber);
+        List<Future<ContainerResponse>> futures = new LinkedList<>();
```

Comment:
Still don't see the point. In this particular case we are adding and iterating. For these operations `LinkedList` and `ArrayList` have the similar complexity, noting that `ArrayList` has constant size of course. 

Not sure I can get what you mean saying 
>  it is all about the heap and objects location in it
unless you provide reliable arguments.

Generally speaking if we talk about optimization I find it quite unusual that you did not mention using java arrays for example. Looks like you are arguing just for a dispute.